packages/plugins/src/cli/install/index.ts

```diff
@@ -14,17 +15,63 @@ export const handleInstall = async () => {
 
   const dependencyInstall = await spawnAsync('forge', args)
   if (dependencyInstall.code !== 0) {
-    // The `stdout` contains the trace of the error.
-    console.log(dependencyInstall.stdout)
-    // The `stderr` contains the error message.
-    console.log(dependencyInstall.stderr)
-    process.exit(1)
+    // If installation fails due to missing tag, then run forge update and attempt again
+    if (
+      dependencyInstall.stderr.includes(
+        `Tag: "${CONTRACTS_LIBRARY_VERSION}" not found for repo "https://github.com/sphinx-labs/sphinx"`
+      )
+    ) {
+      // Run forge install since installing with the incorrect version deletes the previous installation
+      const install = await spawnAsync('forge', ['install'])
+      if (install.code !== 0) {
+        // The `stdout` contains the trace of the error.
+        console.log(install.stdout)
+        // The `stderr` contains the error message.
+        console.log(install.stderr)
+        process.exit(1)
+      }
+
+      // Run forge update to fetch the latest branches
+      const update = await spawnAsync('forge', ['update', 'sphinx-labs/sphinx'])
+      if (update.code !== 0) {
+        // The `stdout` contains the trace of the error.
+        console.log(update.stdout)
+        // The `stderr` contains the error message.
+        console.log(update.stderr)
+        process.exit(1)
+      }
+
+      // Attempt to install the correct version again
+      const dependencyInstallAttemptTwo = await spawnAsync('forge', args)
+      if (dependencyInstallAttemptTwo.code !== 0) {
+        // The `stdout` contains the trace of the error.
+        console.log(dependencyInstallAttemptTwo.stdout)
+        // The `stderr` contains the error message.
+        console.log(dependencyInstallAttemptTwo.stderr)
+        process.exit(1)
+      }
+    } else {
+      // Exit if any other error occurs during first installation
+
+      // The `stdout` contains the trace of the error.
+      console.log(dependencyInstall.stdout)
+      // The `stderr` contains the error message.
+      console.log(dependencyInstall.stderr)
+      process.exit(1)
+    }
   }
+}
+
+export const handleInstall = async (spinner: ora.Ora) => {
+  spinner.start('Installing Sphinx Solidity library...')
+
+  await installWithUpdate()
 
   // Foundry doesn't consistently install our subdependency, so we make sure it's installed ourselves.
   // We should test this out later and remove if possible.
   execSync(
     'cd lib/sphinx/packages/contracts && forge install --no-commit && cd ../../ && git restore packages/contracts/foundry.toml',
     { stdio: 'ignore' }
   )
+  spinner.start('Successfully installed Sphinx Solidity library')
```

Comment:
When I attempt to run `npx sphinx install` in the demo package, the process never exits. I don't know whether that's caused by the fact that we don't call `spinner.success` or if it's caused by the `execSync` call above

Reply:
Good catch, this was related to the fact that we weren't calling `spinner.success`